test/FacadeAct.test.ts

```diff
@@ -200,669 +203,739 @@ describe('FacadeAct contract', () => {
+        await distributor
+          .connect(owner)
+          .setDistribution(FURNACE_DEST, { rTokenDist: 0, rsrDist: 0 })
+        // Transfer free tokens to RTokenTrader
+        const hndAmt: BigNumber = bn('10e18')
+        await rToken.connect(addr1).transfer(rTokenTrader.address, hndAmt)
+
+        let [addr, data] = await facadeAct.callStatic.getActCalldata(rToken.address)
+        expect(addr).to.equal(ZERO_ADDRESS)
+        expect(data).to.equal('0x')
+
+        // RSR can be distributed with no issues - seed stRSR with half as much
+        await rsr.connect(addr1).transfer(backingManager.address, hndAmt)
+        await rsr.connect(addr1).transfer(stRSR.address, hndAmt.div(2))
+        ;[addr, data] = await facadeAct.callStatic.getActCalldata(rToken.address)
+        expect(addr).to.equal(backingManager.address)
+        expect(data).to.not.equal('0x')
+
+        expect(await rsr.balanceOf(stRSR.address)).to.equal(hndAmt.div(2))
+        expect(await rsr.balanceOf(backingManager.address)).to.equal(hndAmt)
+        expect(await rsr.balanceOf(rsrTrader.address)).to.equal(0)
+
+        // Execute managetokens in Backing Manager
+        await addr1.sendTransaction({
+          to: addr,
+          data,
+        })
 
-      // AAVE Rewards - RSR Trader
-      await aToken.setRewards(rsrTrader.address, rewardAmountAAVE)
+        // RSR forwarded
+        expect(await rsr.balanceOf(stRSR.address)).to.equal(hndAmt.add(hndAmt.div(2)))
+        expect(await rsr.balanceOf(backingManager.address)).to.equal(0)
+        expect(await rsr.balanceOf(rsrTrader.address)).to.equal(0)
+      })
 
-      // Via Facade get next call - will claim rewards from Traders, via Facade
-      let [addr, data] = await facadeAct.callStatic.getActCalldata(rToken.address)
-      expect(addr).to.equal(facadeAct.address)
-      expect(data).to.not.equal('0x')
+      it('Should not revert if f=0', async () => {
+        await distributor.connect(owner).setDistribution(STRSR_DEST, { rTokenDist: 0, rsrDist: 0 })
+        // Transfer free tokens to RTokenTrader
+        const hndAmt: BigNumber = bn('10e18')
+        await rsr.connect(addr1).transfer(rsrTrader.address, hndAmt)
 
-      // Claim rewards
-      await owner.sendTransaction({
-        to: addr,
-        data,
-      })
+        let [addr, data] = await facadeAct.callStatic.getActCalldata(rToken.address)
+        expect(addr).to.equal(ZERO_ADDRESS)
+        expect(data).to.equal('0x')
 
-      // Check rewards collected
-      expect(await aaveToken.balanceOf(rsrTrader.address)).to.equal(rewardAmountAAVE)
-      expect(await aaveToken.balanceOf(rTokenTrader.address)).to.equal(bn(0))
+        // RToken can be distributed with no issues
+        await rToken.connect(addr1).transfer(backingManager.address, hndAmt)
+        ;[addr, data] = await facadeAct.callStatic.getActCalldata(rToken.address)
+        expect(addr).to.equal(backingManager.address)
+        expect(data).to.not.equal('0x')
```

Comment:
check function signature

Reply:
I agree that these tests would be better if they also checked the function signature, but `getActCalldata` is not part of the changes in this PR.

Reply:
I didn't add these tests in this PR, I just reorg'd the test file and it made it look like it was all new code.

Reply:
ah ok. i'll make an issue to touch these up though as it seems valuable to test